src/Integration/src/IntegrationBase/IntegrationServices.cs

```diff
@@ -32,7 +36,11 @@ public virtual IMessageBuilderFactory MessageBuilderFactory
             {
                 if (_messageBuilderFactory == null)
                 {
-                    _messageBuilderFactory = _context.GetService<IMessageBuilderFactory>();
+                    _messageBuilderFactory = _context?.GetService<IMessageBuilderFactory>();
+                    if (_messageBuilderFactory == null)
+                    {
+                        _messageBuilderFactory = new DefaultMessageBuilderFactory();
+                    }
```

Comment:
It looks like there are several instances of this pattern in a row, I'll just make this suggestion once to get your reaction: 
```suggestion
                    _messageBuilderFactory = _context?.GetService<IMessageBuilderFactory>() ?? new DefaultMessageBuilderFactory();
```